pkg/distributor/query.go

```diff
@@ -137,11 +137,10 @@ func (d *Distributor) getIngesterReplicationSetsForQuery(ctx context.Context) ([
 	// Lookup ingesters ring because ingest storage is disabled.
 	shardSize := d.limits.IngestionTenantShardSize(userID)
 	r := d.ingestersRing
-
-	// If tenant uses shuffle sharding, we should only query ingesters which are part of the tenant's subring.
-	if lookbackPeriod := d.cfg.ShuffleShardingLookbackPeriod; shardSize > 0 && lookbackPeriod > 0 {
-		r = r.ShuffleShardWithLookback(userID, shardSize, lookbackPeriod, time.Now())
+	if shardSize == 0 {
```

Comment:
This change is not necessary. We could keep querying entire ring, if shuffle-sharding is not enabled -- entire ring contains all read-only and read-write ingesters, which is fine.

Also setting `shardSize` to `r.InstancesCount()` is not necessary -- `ShuffleShardWithLookback` handles `shardSize == 0` already.